client/apps/edit/components/content/sections/text/test/index.test.js

```diff
@@ -0,0 +1,755 @@
+
+      expect(handleBackspace).toBe('not-handled')
+      expect(props.onSetEditing.mock.calls.length).toBe(0)
+    })
+
+    it('Merges section with previous section if at start of block', () => {
+      const component = getWrapper(props)
+      component.instance().onChange(getSelection())
+      component.instance().focus()
+      const handleBackspace = component.instance().handleBackspace({key: 'backspace'})
+
+      expect(handleBackspace).toBe('handled')
+      expect(props.onSetEditing.mock.calls.length).toBe(props.index - 1)
+      expect(component.state().html).toMatch(props.section.body)
+      expect(component.state().html).toMatch(article.sections[props.index - 1].body)
+    })
+  })
+
+  describe('#onPaste', () => {
+    it('Strips or converts unsupported html and linebreaks', () => {
+      const component = getWrapper(props)
+
+      component.instance().onPaste(
+        'Here is a caption about an image yep.',
+        '<meta><script>bad.stuff()</script><h1 class="stuff">Here is a</h1><ul><li><b>caption</b></li><li>about an <pre>image</pre></li></ul><p>yep.</p><br>'
+      )
+      expect(component.state().html).toMatch(
+        '<p>Here is a</p><ul><li><strong>caption</strong></li><li>about an image</li></ul><p>yep.'
+      )
+    })
+
+    it('Does not overwrite existing content', () => {
+      props.section = article.sections[0]
+      const component = getWrapper(props)
+      component.instance().onPaste(
+        'Here is a caption about an image yep.',
+        '<meta><script>bad.stuff()</script><h1 class="stuff">Here is a</h1><ul><li><b>caption</b></li><li>about an <pre>image</pre></li></ul><p>yep.</p><br>'
+      )
+
+      expect(component.state().html).toMatch(
+        '<p>Here is a</p><ul><li><strong>caption</strong></li><li>about an image</li></ul><p>yep.'
+      )
+      expect(component.state().html).toMatch('What would Antoine Court de Gébelin think of the Happy Squirrel?')
+    })
+
+    it('Strips google inserted styles', () => {
+      props.section.body = ''
+      const component = getWrapper(props)
+      component.instance().onPaste(
+        '<b style="font-weight:normal;" id="docs-internal-guid-ce2bb19a-cddb-9e53-cb18-18e71847df4e"><p><span style="font-size:11pt;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre-wrap;">Available at: Espacio Valverde • Galleries Sector, Booth 9F01</span></p>'
+      )
+      expect(component.state().html).toBe('<p>Available at: Espacio Valverde • Galleries Sector, Booth 9F01</p>')
+    })
+
+    it('Calls #standardizeSpacing', () => {
+      props.section.body = ''
+      const badHtml = '<p>Hello.</p><h2></h2><h3></h3><p></p><p></p>'
+      const component = getWrapper(props)
+      component.instance().onPaste(badHtml)
+
+      expect(component.state().html).toBe('<p>Hello.</p>')
+    })
+  })
+})
+
+// xdescribe '#handleChangeSection', ->
+
```

Comment:
somehow missed these -- will add later today

Reply:
👍  I trust you have a good sense of the coverage in this file, but on first glance it seems like it's missing some tests like `handleKeyCommand` and the stuff that happens in `componentDidUpdate`? All this to say, I think this would be much simpler to evaluate if these were broken out somehow!

Reply:
HandleKeyCommand is here - and further in the keybindings utils tests
https://github.com/artsy/positron/pull/1569/files#diff-09414ab6a2a89e0a22eeff51a9f2077cR342

Reply:
The tests are the same as the coffee version, with new ones added for things that were moved to utils. Could be more granular is true. 
Can take a look at using separate test files for the different describe blocks.